jdaviz/configs/default/plugins/model_fitting/fitting_backend.py

```diff
@@ -100,7 +100,13 @@ def _fit_1D(initial_model, spectrum, run_fitter, window=None):
 
     """
     if run_fitter:
-        weights = 'unc' if spectrum.uncertainty else None
+        if spectrum.uncertainty:
+            if np.all(spectrum.uncertainty.array == 0):
+                weights = None
+            else:
+                weights = 'unc'
```

Comment:
I am not comfortable with `==`. Any number close enough to zero would give you that error.
```suggestion
        if spectrum.uncertainty and not np.allclose(spectrum.uncertainty.array, 0):
            weights = 'unc'
```

Reply:
Would it? How small does `x` have to be for `np.isfinite(1/x)` to return `False`?

Reply:
As far as I can tell, any number that is able to be represented as non-zero in python passes the `np.isfinite(1/x)` test.

Reply:
I think I would be more worried about filtering out uncertainties smaller than `atol` in `np.allclose`.

Reply:
Rare case but not impossible. Might also be machine dependent.

```python
>>> import numpy as np
>>> 1 / 1e-309
inf
>>> 1e-309 == 0
False
>>> np.allclose(1e-309, 0)
True
```

Reply:
Even if you insist on `==`, the nested `if` is unnecessary?

Reply:
I think that for some reason I thought that would be too long for a single line? No idea, I'll fix it 😆